src/plugins/instrument_fetch.js

```diff
@@ -158,42 +158,46 @@ class InstrumentFetch {
 
         return function (request, options = {}) {
             request = new Request(request, options);
-            const url = request.url;
             const opts = tracer.options();
 
-            if (!self._shouldTrace(tracer, url)) {
+            if (!self._shouldTrace(tracer, request.url)) {
                 return proxiedFetch.apply(null, arguments);
             }
 
             let span = tracer.startSpan('fetch');
             tracer.addActiveRootSpan(span);
 
+            const parsed = new URL(request.url);
             let tags = {
-                method : options && options.method ? options.method : 'GET',
-                url    : url,
-            };
-            if (url) {
-                tags.url_pathname = url.split('?')[0];
-            }
-
-            const fetchPayload = Object.assign({}, tags);
+                method : request.method,
+                url    : request.url,
 
+                // NOTE: purposefully excluding username:password URL properties.
+                hash     : parsed.hash,
+                href     : parsed.href,
+                protocol : parsed.protocol,
+                origin   : parsed.origin,
+                host     : parsed.host,
+                hostname : parsed.hostname,
+                port     : parsed.port,
+                pathname : parsed.pathname,
+                search   : parsed.search,
+            };
             if (opts.include_cookies) {
-                fetchPayload.cookies = getCookies();
+                tags.cookies = getCookies();
             }
 
             // Add Open-Tracing headers
             const headersCarrier = {};
             tracer.inject(span.context(), opentracing.FORMAT_HTTP_HEADERS, headersCarrier);
-            const keys = Object.keys(headersCarrier);
-            keys.forEach((key) => {
+            Object.keys(headersCarrier).forEach((key) => {
                 if (!request.headers.get(key)) request.headers.set(key, headersCarrier[key]);
             });
             span.log({
                 event       : 'sending',
-                method      : options.method || 'GET',
-                url         : url,
-                openPayload : fetchPayload,
+                method      : request.method,
+                url         : request.url,
+                openPayload : tags,
```

Comment:
This used to be `fetchPayload`. But the `fetchPayload` wasn't actually sent to the `fetch`. Hopefully this simplification is ok and I didn't miss some nuance? Seemed unnecessary to copy the object when the `tags` object was only ever used for `span.addTags`

Reply:
I think we were following the convention that was set in the `xhr_instrumentation` file and the difference between both the tags and fetchpayload is that tags is only set in the tags and they don't include the possible cookies. 

I'm unsure as to the reasons why fetchpayload shouldn't be used for both the logs and tags

Reply:
Cool, I'll leave as is then!